src/DurableTask.AzureStorage/Partitioning/TablePartitionManager.cs

```diff
@@ -0,0 +1,593 @@
+
+                    if (partition.CurrentOwner == workerName)
+                    {
+                        leaseNum++;
+                        ETag etag = partition.ETag;
+                        
+                        if (partition.IsDraining)
+                        {
+                            if (tasks[partition.RowKey!].IsCompleted == true)
+                            {
+                                ReleaseLease(partition);
+                                isReleasedLease = true;
+                                leaseNum--;
+                            }
+                            else
+                            {
+                                RenewLease(partition);
+                            }
+                        }
+                        else
+                        {
+                            DrainLease(partition, CloseReason.Shutdown);
+                            RenewLease(partition);
+                            isDrainedLease = true;
+                        }
+                        
+                        try
+                        {
+                            await this.myTable.UpdateEntityAsync(partition, etag, (TableUpdateMode)1);
+                            if (isDrainedLease)
+                            {
+                                this.settings.Logger.PartitionManagerInfo(
+                                     this.storageAccountName,
+                                     this.settings.TaskHubName,
+                                     this.settings.WorkerId,
+                                     partition.RowKey,
+                                     $"Worker {this.settings.WorkerId} starts draining the lease of{partition.RowKey}.");
+                            }
+                            if (isReleasedLease)
+                            {
+                                this.settings.Logger.PartitionManagerInfo(
+                                     this.storageAccountName,
+                                     this.settings.TaskHubName,
+                                     this.settings.WorkerId,
+                                     partition.RowKey,
+                                     $"Worker {this.settings.WorkerId} releases the lease of {partition.RowKey}.");
+                            }
+                            
+                        }
+                        //Exception will be thrown if update operation fails.
+                        //Need to re-read the table again to get the latest ETag.
+                        catch (RequestFailedException ex) when (ex.Status == 412)
+                        {
+                            throw ex;
+                        }
+                    }
+
+                }
+                if (leaseNum == 0)
+                {
+                    return true;
+                }
+                else
+                {
+                    return false;
+                }
```

Comment:
I think we can simplify this as follows:

```suggestion
                var isLeaseNumZero = (leaseNum == 0);
                return isLeaseNumZero
```